numpy/ma/core.py

```diff
@@ -1481,9 +1481,14 @@ def getmaskarray(arr):
     """
     mask = getmask(arr)
     if mask is nomask:
-        mask = make_mask_none(np.shape(arr), getattr(arr, 'dtype', None))
+        mask = make_mask_none(
+            np.shape(arr),
+            dtype=getattr(arr, 'dtype', None),
+            readonly=nomask_as_readonly)
     return mask
 
+def _viewmaskarray(arr):
```

Comment:
The problem I have with that is that `_viewmaskarray(arr, writable=False).writable == False` does not always hold, which might be a little surprising.